build.sh

```diff
@@ -38,6 +38,8 @@ export LDFLAGS="-L$TARGET/lib -O3 -sEXPORTED_FUNCTIONS=_main,_malloc,_free -sALL
 if [ "$WASM_BIGINT" = "true" ]; then
   export LDFLAGS+=" -sWASM_BIGINT"
 fi
+# Ensure the `ffi_call` symbol can be found at link time
+export LDFLAGS+=" -sLLD_REPORT_UNDEFINED=0"
```

Comment:
This is an unfortunate thing to have to add.  

Why is this symbol defined?  Is it and EM_JS symbol?  Do you know where the undefined reference coming from?

Reply:
Good question. It is defined here:
https://github.com/hoodmane/libffi-emscripten/blob/8bcb4262bfc8cb57b6314b93143f18c650f63492/src/wasm32/ffi.c#L173
(i.e., as an `EM_JS` symbol)

> Do you know where the undefined reference coming from?

I'm not sure, let me investigate that. Perhaps defining an `ffi_call` stub in native code fixes this.

Reply:
... relevant header declaration:
https://github.com/hoodmane/libffi-emscripten/blob/8bcb4262bfc8cb57b6314b93143f18c650f63492/include/ffi.h.in#L482-L486

Reply:
I think I see what is happening.  EM_JS functions are functions that are defined in native code.  We they kind of need to be undefined when wasm-ld runs, so it knows to generate imports for these symbols.

When an EM_JS functions is used within the same file where it is declared this is not an issues since the macro declares the symbol with an `import_name` attribute, which tells the linker its OK to be undefined/imported.

As you suggested I think the simplest solution would be write a native wrapper for the EM_JS function in the same file where it is defined.

Reply:
Native wrapper function implemented with commit d1798d88fb955900b2452af3c28f487fa85b6808, seems to work!